traitsui/tests/test_editor.py

```diff
@@ -917,3 +917,54 @@ def test_sync_value_both(self):
 
         with self.assertTraitDoesNotChange(user_object, "user_auxiliary"):
             editor.auxiliary_value = 14
+
+    def test_editor_error_msg(self):
+        from pyface.toolkit import toolkit_object
+        from pyface.qt import QtGui
+        from traits.api import HasTraits, Range
+
+        from traitsui.testing.api import (
+             KeyClick, KeySequence, Textbox, UITester
+        )
+
+        ModalDialogTester = toolkit_object(
+            "util.modal_dialog_tester:ModalDialogTester"
+        )
+
+        class Foo(HasTraits):
+            x = Range(low=0.0, high=1.0, value=0.5, exclude_low=True)
+
+        foo = Foo()
+        tester = UITester()
+        with tester.create_ui(foo) as ui:
+
+            x_range = tester.find_by_name(ui, "x")
+            x_range_textbox = x_range.locate(Textbox())
+
+            for _ in range(3):
+                x_range_textbox.perform(KeyClick('Backspace'))
+
+            x_range_textbox.perform(KeySequence('0.0'))
+
+            def trigger_error():
+                x_range_textbox.perform(KeyClick('Enter'))
+
+            def check_and_close(mdtester):
+                try:
+                    with mdtester.capture_error():
+                        self.assertTrue(
+                            mdtester.has_widget(
+                                text="The 'x' trait of a Foo instance must be "
+                                     "0.0 < a floating point number <= 1.0, "
+                                     "but a value of 0.0 <class 'float'> was "
+                                     "specified.",
+                                type_=QtGui.QMessageBox,
+                            )
+                        )
+                finally:
+                    mdtester.close(accept=True)
+                    self.assertTrue(mdtester.dialog_was_opened)
+
+            mdtester = ModalDialogTester(trigger_error)
+            mdtester.open_and_run(check_and_close)
+            self.assertTrue(mdtester.dialog_was_opened)
```

Comment:
The level of complexity for this test now seems to far exceed what it is trying to test :/
Perhaps the test could be trivia and just reach into Qt and make sure the `textFormat` property gets set? Like I do in https://github.com/enthought/pyface/pull/907

The macOS weirdness I am seeing feels like an orthogonal problem, and I am still very unclear on what the cause is.

Reply:
> The macOS weirdness I am seeing feels like an orthogonal problem, and I am still very unclear on what the cause is.

Yes. Definitely something strange going on here.